protocol/protocol.go

```diff
@@ -31,194 +46,226 @@ func (m *Manager) routerInitializer(router *routerpkg.Router, netConnection *net
-		},
-	)
+func (m *Manager) addIBDFlows(router *routerpkg.Router, isStopping *uint32, stop chan error) []*flow {
 
-	m.addFlow("HandleGetBlockLocator", router, []wire.MessageCommand{wire.CmdGetBlockLocator}, isStopping, stop,
-		func(incomingRoute *routerpkg.Route) error {
-			return ibd.HandleGetBlockLocator(m.context, incomingRoute, outgoingRoute)
-		},
-	)
+	outgoingRoute := router.OutgoingRoute()
 
-	m.addFlow("HandleGetBlocks", router, []wire.MessageCommand{wire.CmdGetBlocks}, isStopping, stop,
-		func(incomingRoute *routerpkg.Route) error {
-			return ibd.HandleGetBlocks(m.context, incomingRoute, outgoingRoute)
-		},
-	)
+	return []*flow{
+		m.addFlow("HandleIBD", router, []wire.MessageCommand{wire.CmdBlockLocator, wire.CmdIBDBlock}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return ibd.HandleIBD(m.context, incomingRoute, outgoingRoute, peer)
+			},
+		),
+
+		m.addFlow("RequestSelectedTip", router, []wire.MessageCommand{wire.CmdSelectedTip}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return selectedtip.RequestSelectedTip(m.context, incomingRoute, outgoingRoute, peer)
+			},
+		),
+
+		m.addFlow("HandleGetSelectedTip", router, []wire.MessageCommand{wire.CmdGetSelectedTip}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return selectedtip.HandleGetSelectedTip(m.context, incomingRoute, outgoingRoute)
+			},
+		),
+
+		m.addFlow("HandleGetBlockLocator", router, []wire.MessageCommand{wire.CmdGetBlockLocator}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return ibd.HandleGetBlockLocator(m.context, incomingRoute, outgoingRoute)
+			},
+		),
+
+		m.addFlow("HandleGetBlocks", router, []wire.MessageCommand{wire.CmdGetBlocks}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return ibd.HandleGetBlocks(m.context, incomingRoute, outgoingRoute)
+			},
+		)}
 }
 
-func (m *Manager) addTransactionRelayFlow(router *routerpkg.Router, isStopping *uint32, stop chan error) {
-
+func (m *Manager) addTransactionRelayFlow(router *routerpkg.Router, isStopping *uint32, stop chan error) []*flow {
 	outgoingRoute := router.OutgoingRoute()
 
-	m.addFlow("HandleRelayedTransactions", router, []wire.MessageCommand{wire.CmdInv, wire.CmdTx}, isStopping, stop,
-		func(incomingRoute *routerpkg.Route) error {
-			return relaytransactions.HandleRelayedTransactions(m.context, incomingRoute, outgoingRoute)
-		},
-	)
+	return []*flow{
+		m.addFlow("HandleRelayedTransactions", router, []wire.MessageCommand{wire.CmdInv, wire.CmdTx}, isStopping, stop,
+			func(incomingRoute *routerpkg.Route, peer *peerpkg.Peer) error {
+				return relaytransactions.HandleRelayedTransactions(m.context, incomingRoute, outgoingRoute)
+			},
+		)}
 }
 
 func (m *Manager) addFlow(name string, router *routerpkg.Router, messageTypes []wire.MessageCommand, isStopping *uint32,
```

Comment:
addFlow is a misleading name because it returns a new flow.
You can leave the name if you keep a shared `manager.flows`